Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/Shadow/ProjectedShadow.azsli

```diff
@@ -145,72 +142,29 @@ float ProjectedShadow::GetVisibilityNoFilter()
 
 float ProjectedShadow::GetVisibilityPcf()
 {
-    // PCF filter method is stored in bottom 16 bits.
-    const uint pcfFilterMethod = ViewSrg::m_projectedShadows[m_shadowIndex].m_shadowFilterMethod >> 16;
-    if (pcfFilterMethod == PcfFilterMethod_Bicubic)
-    {
-        return SamplePcfBicubic();
-    }
-
-    const uint predictionCount = ViewSrg::m_projectedShadows[m_shadowIndex].m_predictionSampleCount;
-
-    if (predictionCount <= 1)
-    {
-        return GetVisibilityNoFilter();
-    }
-
-    const float4 jitterUnitVectorDepthDiffBase =
-        Shadow::GetJitterUnitVectorDepthDiffBase(m_normalVector, m_lightDirection);
-    const float3 jitterUnitY = jitterUnitVectorDepthDiffBase.xyz;
-    const float3 jitterUnitX = cross(jitterUnitY, m_lightDirection);
-    const float jitterDepthDiffBase = jitterUnitVectorDepthDiffBase.w;
+    const uint filteringSampleCount = ViewSrg::m_projectedShadows[m_shadowIndex].m_filteringSampleCount;
+    const float3 atlasPosition = GetAtlasPosition(m_shadowPosition.xy);   
 
-    uint shadowedCount = 0;
-    uint jitterIndex = 0;
+    SampleShadowMapBicubicParameters param;
+    param.shadowMap = PassSrg::m_projectedShadowmaps;
+    param.shadowPos = float3(atlasPosition.xy * ViewSrg::m_invShadowmapAtlasSize, atlasPosition.z);
+    param.shadowMapSize = ViewSrg::m_shadowmapAtlasSize;
+    param.invShadowMapSize = ViewSrg::m_invShadowmapAtlasSize;
+    param.comparisonValue = m_shadowPosition.z - m_bias;
+    param.samplerState = SceneSrg::m_hwPcfSampler;
 
-    // Predicting
-    for (; jitterIndex < predictionCount; ++jitterIndex)
-    {
-        if (IsShadowedWithJitter(
-            jitterUnitX,
-            jitterUnitY,
-            jitterDepthDiffBase,
-            jitterIndex))
-        {
-            ++shadowedCount;
-        }
-    }
-    if (shadowedCount == 0)
+    if (filteringSampleCount <= 4)       
     {
-        return 1.;
+        return SampleShadowMapBicubic_4Tap(param);
     }
-    else if (shadowedCount == predictionCount)
+    else if (filteringSampleCount <= 9)
     {
-        return 0.;
+        return SampleShadowMapBicubic_9Tap(param);
     }
-
-    // Filtering
-
-    // When the prediction detects the point on the boundary of shadow,
-    // i.e., both of a lit point and a a shadowed one exists in the jittering area,
-    // we calculate the more precious lit ratio in the area.
-    const uint filteringCount = max(
-        predictionCount,
-        ViewSrg::m_projectedShadows[m_shadowIndex].m_filteringSampleCount);
-
-    for (; jitterIndex < filteringCount; ++jitterIndex)
+    else
     {
-        if (IsShadowedWithJitter(
-            jitterUnitX,
-            jitterUnitY,
-            jitterDepthDiffBase,
-            jitterIndex))
-        {
-            ++shadowedCount;
-        }
+        return SampleShadowMapBicubic_16Tap(param);
     }
-
-    return (filteringCount - shadowedCount) * 1. / filteringCount;
 }
```

Comment:
Any chance we could do a thread ballot here and pick the max value used across the threadgroup? Otherwise we're doing the sum of all active choices.